protobuf/v1/nexus.proto

```diff
@@ -0,0 +1,129 @@
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+import "common.proto";
+
+package mayastor.v1;
+
+service NexusRpc {
+  // Nexus related methods.
+  //
+  // Nexus is a logical frontend representing a data volume taking care of
+  // replication and rebuild in the background.
+
+  rpc CreateNexus (CreateNexusRequest) returns (Nexus) {}
```

Comment:
If a CreateNexus fails at the moment we have no way of knowing why, eg: was it a child that was inaccessible? We simply return a grpc status Internal AFAIK
Instead of returning the Nexus could we return a oneOf of either Nexus or a well defined error (which would indicate the reason and the children that could not be connected to for example..)?

Reply:
This would mean that the `rpc` call always returns a `OK` status instead of error even when the call fails. Better approach would be to introduce the new error struct with details about all children of nexus in the status message itself which can be parsed at the control plane to take actions accordingly.